src/main/Application.java

```diff
@@ -21,10 +23,10 @@ public static final Application getInstance() {
 
     public static final Path assetsDir = Path.of("assets");
 
-    private Application() {
+    private Application(Database db) {
         instance = this;
         init();
-        setPanel(new LoginPanel());
+        setPanel(new LoginPanel(db));
```

Comment:
then would be no need to pass it to those pannels as we could retrieve it  from app